### PR TITLE
Update DyNAS-T to 1.5

### PR DESCRIPTION
**Pull request by macsz**
## Type of Change

Update DyNAS-T to 1.5

## Description

Quantization- and architecture-aware search are one of the new features in DyNAS-T 1.5. 

This approach allows you to run search on your FP32 super-network and find optimal model configurations w.r.t. both architecture and Post-Training Quantization policy. DyNAS-T's implementation uses [Intel® Neural Compressor](https://github.com/intel/neural-compressor) as an underlying backend for quantizing models.

Please refer to the notebook in this PR for example of usage.

Additional changes:

* in tests `config.dynas.batch_size` has been reduced to 1 to speedup testing. Latency-based measurement is done over number of steps and while the number of steps gets scaled up automatically for smaller batch sizes by DyNAS-T automatically, it's still faster for testing purposes to use batch size 1 for latency measurement.
* Configuration is updated with two new parameters:
    * `eval_batch_size` - can be used to change batch sized used during forward pass in accuracy evaluation. It detaches this configuration option from batch size used for latency measurement.
    * `test_fraction` - Fraction (0.0, 1.0> of the validation data to be used for testing and evaluation. Using smaller values can significantly accelerate search process, but might also introduce noise to the accuracy measurement.

## How has this PR been tested?

DyNAS-T has unit tests and each release undergoes functional tests to verify search results. Test suite is available [here](https://github.com/IntelLabs/DyNAS-T/tree/main/tests).

## Dependency Change?

`dynast==1.3.0` changed to `dynast==1.5.1`

### CODE REVIEW

File: examples/notebook/dynas/ResNet50_Quantiation_Search_Supernet_NAS.ipynb [hunk 1]
```diff
@@ -0,0 +1,481 @@
...
+      "2023-09-05 14:10:32 [INFO] |        Linear       |   1   |  0   |  1   |\n",
+      "2023-09-05 14:10:32 [INFO] +---------------------+-------+------+------+\n",
+      "2023-09-05 14:10:32 [INFO] Pass quantize model elapsed time: 9604.32 ms\n",
+      "2023-09-05 14:10:32 [INFO] Save tuning history to /localdisk/maciej/code/neural-compressor/examples/notebook/dynas/nc_workspace/2023-09-05_14-07-22/./history.snapshot.\n",
+      "2023-09-05 14:10:32 [INFO] Specified timeout or max trials is reached! Found a quantized model which meet accuracy goal. Exit.\n",
+      "2023-09-05 14:10:32 [INFO] Save deploy yaml to /localdisk/maciej/code/neural-compressor/examples/notebook/dynas/nc_workspace/2023-09-05_14-07-22/deploy.yaml\n"
+     ]
+    },
+    {
+     "ename": "KeyboardInterrupt",
```
Comment (XinyuYe-Intel) [inline examples/notebook/dynas/ResNet50_Quantiation_Search_Supernet_NAS.ipynb:324]: Seems notebook haven't run successfully, I think it's better to have a successful run.

Comment (macsz) [reply examples/notebook/dynas/ResNet50_Quantiation_Search_Supernet_NAS.ipynb]: Oops! Good catch - commit was created without staging changes in the notebook. Thanks!

Comment (XinyuYe-Intel) [reply examples/notebook/dynas/ResNet50_Quantiation_Search_Supernet_NAS.ipynb]: No problem.

**Review by XinyuYe-Intel — APPROVED** [review]

**Review by chensuyue — APPROVED** [review]

**Pull request merged**